integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/model/JpaColumnScaleMappingIT.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.model;
+
+import java.math.BigDecimal;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.ScaledNumberField;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+public class JpaColumnScaleMappingIT {
+
+	private static final String INDEX_NAME = "IndexName";
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
+
+	@Test
+	public void mapping() {
+
+		backendMock.expectSchema( INDEX_NAME, b -> b
+				.field( "scaled", BigDecimal.class )
+				.field( "unscaled", BigDecimal.class )
+				.field( "defaultScaled", BigDecimal.class )
+				.field( "hsearch", BigDecimal.class, f -> f.decimalScale( 7 ) )
+		);
+
+		ormSetupHelper.withBackendMock( backendMock ).setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Entity(name = "indexed")
+	@Indexed(index = INDEX_NAME)
+	public static final class IndexedEntity {
+
+		@Id
+		@DocumentId
+		private Integer id;
+
+		@ScaledNumberField
+		@Column(precision = 14, scale = 7)
+		private BigDecimal scaled;
```

Comment:
Shouldn't it work with `@GenericField`, too? I would expect it to.

Reply:
Yep. There is a test for it.

Reply:
As discussed, there is not. To do.

Reply:
Added and you were right. It already works with `@GenericField`, since the subtype of the context depends on field type and not on the annotation we use.